src/server/.tests/ws-routes/controllers/Room.spec.js

```diff
@@ -0,0 +1,395 @@
+const assert = require('chai').assert
+const _ = require('lodash')
+const sinon = require('sinon')
+const socketClient = require('socket.io-client')
+const io = require('socket.io')
+const RoomProvider = require('../../../ws-routes/controllers/Room')
+const config = require('../../../config.json')
+const namespaceInfo = require('../../../protocol/protocol.json').room
+
+
+const SERVER_EVENTS = namespaceInfo.serverEventTypes
+const CLIENT_EVENTS = namespaceInfo.eventTypes
+const SOCKET_PORT = config.httpServer.port
+const SOCKET_URL = `http://localhost:${SOCKET_PORT}`
+const SOCKET_OPTIONS = {
+	transports: ['websocket'],
+	'force new connection': true
+}
+
+let suite
+describe('Room websocket service', () => {
+	beforeEach(() => {
+		suite = {}
+		suite.connectionsMock = {
+			usersToConnectionsMap: new Map()
+		}
+		suite.server = io.listen(SOCKET_PORT)
+		suite.USERNAME_MOCK = 'DUMMY_USERNAME'
+		suite.middlewareMock = sinon.stub().callsFake((socket, next) => {
+			socket.request.user = {username: suite.USERNAME_MOCK}
+			next()
+		})
+		suite.server.use(suite.middlewareMock)
+		suite.client = {}
+	})
+
+	afterEach(done => {
+		suite.server.close(done)
+		if (_.isFunction(suite.client.disconnect)) suite.client.disconnect()
+	})
+
+	describe('#connection', () => {
+		it('should send joined event when connected', done => {
+			//given
+			suite.server.on(CLIENT_EVENTS.CONNECTION, socket => RoomProvider.connection(socket, suite.connectionsMock))
+
+			//when
+			suite.client = socketClient.connect(SOCKET_URL, SOCKET_OPTIONS)
+
+			//then
+			suite.client.on(SERVER_EVENTS.JOINED, then)
+			function then(data) {
+				done()
+			}
+		})
+
+		it('should new entry be added with its key as user name to connections repository when connected', done => {
+			//given
+			suite.server.on(CLIENT_EVENTS.CONNECTION, socket => RoomProvider.connection(socket, suite.connectionsMock))
+
+			//when
+			suite.client = socketClient.connect(SOCKET_URL, SOCKET_OPTIONS)
+
+			//then
+			suite.client.on(SERVER_EVENTS.JOINED, then)
+
+			function then(data) {
+				const hasSessionStored = suite.connectionsMock.usersToConnectionsMap.has(suite.USERNAME_MOCK)
+				assert.isTrue(hasSessionStored)
+				done()
+			}
+		})
+
+		it('should add a new entry to connections list which is instance of socket when connecting to Room namespace', done => {
+			//given
+			
```

Comment:
Uhh, what type of human are you ? You left unnecessary blank line.

Reply:
oka